### PR TITLE
Update Tiltfile to use env variable to check aks cluster is being used

### LINKED ISSUE #5289: Update Tiltfile to differentiate between aks and kind cluster
Tiltfile uses `if "aks" in settings.get("kustomize_substitutions", {}).get("MGMT_CLUSTER_NAME", ""):` to check if the tilt is being used with AKS cluster as a management cluster. 
This is an inefficient approach. Especially when a user is switching from AKS -> KIND. 

Suggested approach:
- There are two approaches to solve this
  1. Have `MGMT_CLUSTER_TYPE` saved to an .env file in the repo directory with `MGMT_CLUSTER_TYPE=aks` when running `make aks-create`. Source this env file before running `tilt-up`. The env file must be updated when running `make kind-create`. 
  **OR**
  2. Update **tilt-settings.yaml**, add or delete `MGMT_CLUSTER_TYPE` every time `aks-create` or `kind-create` is called. Tilt-settings.yaml gets parsed in Tiltfile https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/6bce7f87c9ae8c49a3e352683daacbc209127248/Tiltfile#L41
    - Once settings struct in tiltfile is updated, we can then use it to check if the mgmt cluster type is aks or not. 
- Once a similar logic is implemented, we want to update all the occurrences of `if "aks" in settings.get("kustomize_substitutions", {}).get("MGMT_CLUSTER_NAME", ""):` in the Tiltfile with appropriate if condition.

### PR DESCRIPTION
**Pull request by rihib**
**What type of PR is this?**
kind cleanup

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #5289

**Special notes for your reviewer**:

- [ ] cherry-pick candidate

**TODOs**:

- [x] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note

```

### CODE REVIEW

**Comment by rihib** [comment]
cc. @nawazkh

**Review by mboersma — APPROVED** [review]
/lgtm
/assign @nawazkh

File: Tiltfile [hunk 1]
```diff
@@ -50,8 +50,9 @@ if "allowed_contexts" in settings:
 if "default_registry" in settings:
     default_registry(settings.get("default_registry"))
 
+mgmt_cluster_type = os.getenv("MGMT_CLUSTER_TYPE")
```
Comment (nawazkh) [inline Tiltfile:53]:
```suggestion
mgmt_cluster_type = os.getenv("MGMT_CLUSTER_TYPE", "")
```

**Review by nawazkh — COMMENTED** [review]
I am sorry for the delay, this looks good to me pending one suggestion.

**Comment by rihib** [comment]
@nawazkh Thank you for the review! I have made corrections based on your feedback.

**Review by mboersma — APPROVED** [review]
/lgtm

**Review by mboersma — APPROVED** [review]
/approve

**Comment by nawazkh** [comment]
I apologize for lgtm-ing this PR. This fix does not work, we need to revert this change. 

This change does not work because each Make target runs in its own sub-shell, setting MGMT_CLUSTER_TYPE in one target does not automatically make it available in another target.

**Comment by nawazkh** [comment]
This is really my mistake, I am sorry @rihib . The author was following my directions put up in the issue #5289 . 

The right fix would be to update the exported env variables in tilt-settings.yaml or have an another env file where `MGMT_CLUSTER_TYPE` can be saved and sourced right before running Tiltfile. 

@rihib , if you are still up for it, could you please experiment with saving `MGMT_CLUSTER_TYPE` env file and source it in the tilt-up` recipe ? 
I think the original idea of exporting and unsetting in `aks-create` and `kind-create` are still valid, we have to set the `MGMT_CLUSTER_TYPE` in the env file instead. 

I will continue my thought in the issue.

**Comment by rihib** [comment]
@nawazkh Got it, thanks for the clarification! I’ll give it a try.

**Pull request merged**